oe_authentication.module

```diff
@@ -10,10 +10,19 @@ declare(strict_types=1);
 /**
  * Implements hook_user_cancel_methods_alter().
  *
- * Remove the option to delete users while keeping the option to block them.
+ * Conditionally removes the option to delete users.
  */
 function oe_authentication_user_cancel_methods_alter(&$methods) {
-  if (\Drupal::currentUser()->id() == 1) {
+  $current_user = \Drupal::currentUser();
+  $restrict_user_delete = \Drupal::configFactory()
+    ->get('oe_authentication.settings')
+    ->get('restrict_user_delete_cancel_methods');
+  // @todo The restriction should not check for UID 1 as it's bad practice. It
+  // should either check for the user being an admin, or use a permission based
+  // approach like the module drupal/user_cancel_methods_permissions.
+  // Furthermore, the 'access' property of the cancel methods should be used,
+  // rather than removing the method altogether.
```

Comment:
Lines after the first need to be indented
```suggestion
  // @todo The restriction should not check for UID 1 as it's bad practice. It
  //   should either check for the user being an admin, or use a permission based
  //   approach like the module drupal/user_cancel_methods_permissions.
  //   Furthermore, the 'access' property of the cancel methods should be used,
  //   rather than removing the method altogether.
```

Reply:
I see it has been indented now.